### PR TITLE
docs(website): Add docsearch migration blog post

### PR DESCRIPTION
**Pull request by slorber**
## Motivation

Algolia DocSearch is migrating.

After 1st of Feb 2022, all sites should migrate to this new system and have updated their site config.

Review: https://deploy-preview-5967--docusaurus-2.netlify.app/blog/2021/11/18/algolia-docsearch-migration/

### CODE REVIEW

File: website/blog/2021-11-18-algolia-docsearch-migration/index.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    url: https://github.com/shortcuts
+    image_url: https://github.com/shortcuts.png
+    twitter: sh0rtcts
+  - key: slorber
+
+tags: [search]
+image: ./img/social-card.png
+---
+
+[DocSearch](https://docsearch.algolia.com/) is migrating to a new, more powerful system.
```
Comment (shortcuts) [inline website/blog/2021-11-18-algolia-docsearch-migration/index.md:15]:
```suggestion
[DocSearch](https://docsearch.algolia.com/) is migrating to a new, more powerful system, which gives users their own Algolia application and new credentials.
```

I think it's best to state it early so they know why they should migrate

File: website/blog/2021-11-18-algolia-docsearch-migration/index.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+Docusaurus site owners should upgrade their site configuration **by February 1, 2022**
+
+Existing search indexes will be frozen and become read-only state.
```
Comment (shortcuts) [inline website/blog/2021-11-18-algolia-docsearch-migration/index.md:19]:
```suggestion
Docusaurus site owners should upgrade their configuration with [their new credentials](#im-using-docusaurus-and-docsearch-can-i-migrate) **by February 1, 2022**, existing search indexes will be frozen and become read-only state after this date.
```

This seems less "critical" this way

File: website/blog/2021-11-18-algolia-docsearch-migration/index.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+<!--truncate-->
+
+## Upgrading your Docusaurus site
+
+In the next few weeks, Docusaurus site owners will receive an email inviting them to join an Algolia application.
```
Comment (shortcuts) [inline website/blog/2021-11-18-algolia-docsearch-migration/index.md:25]:
```suggestion
In the next few weeks, Docusaurus site owners will receive an email inviting them to join their personal Algolia application.
```

**Review by shortcuts — COMMENTED** [review]
General feedbacks, looks good overall!

File: website/blog/2021-11-18-algolia-docsearch-migration/index.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+### What is DocSearch?
+
+[DocSearch](https://docsearch.algolia.com/) is a program created by [Algolia](http://algolia.com/), which offers search to technical documentation of Open Source projects and technical blogs **for free**.
+
+You can [read more here](https://docsearch.algolia.com/docs/what-is-docsearch/), and [apply](https://docsearch.algolia.com/apply) if you'd like to give it a try!
+
+## What is this migration about?
+
+### Motivations
```
Comment (JoelMarcey) [inline website/blog/2021-11-18-algolia-docsearch-migration/index.md:67]:
```suggestion
### Motivation
```

File: website/blog/2021-11-18-algolia-docsearch-migration/index.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+### Where can I read more about this?
+
+We wrote a small [migration guide](https://docsearch.algolia.com/docs/migrating-from-legacy) but you'll have more detailed information in the migration email.
+
+### I received the migration email but the invite expired
+
+Please contact us via either [email](mailto:docsearch@algolia.com) or [DocSearch's Discord](https://discord.gg/bRTacwYrfX) or on [Docusaurus's Discord #algolia channel](https://discordapp.com/invite/docusaurus)
+
+### I have feedbacks!
```
Comment (JoelMarcey) [inline website/blog/2021-11-18-algolia-docsearch-migration/index.md:109]:
```suggestion
### I have feedback!
```

**Review by JoelMarcey — APPROVED** [review]
with a few nits

File: website/blog/2021-11-18-algolia-docsearch-migration/index.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    twitter: sh0rtcts
+  - key: slorber
+
+tags: [search]
+image: ./img/social-card.png
+---
+
+[DocSearch](https://docsearch.algolia.com/) is migrating to a new, more powerful system.
+
+Docusaurus site owners should upgrade their site configuration **by February 1, 2022**
```
Comment (bodinsamuel) [inline website/blog/2021-11-18-algolia-docsearch-migration/index.md:17]: Not sure it's worth it but still suggesting.
```suggestion
Docusaurus site owners with Docsearch enabled should upgrade their site configuration **by February 1, 2022**
```

File: website/blog/2021-11-18-algolia-docsearch-migration/index.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+- Join the Algolia application (eventually creating a new Algolia account)
+- Update your site configuration.
+
+```js title="docusaurus.config.js"
+const config = {
+  themeConfig: {
+    algolia: {
+      // highlight-start
+      appId: '<NEW_APP_ID>',
+      apiKey: '<NEW_API_KEY>',
```
Comment (bodinsamuel) [inline website/blog/2021-11-18-algolia-docsearch-migration/index.md:40]:
```suggestion
      apiKey: '<NEW_SEARCH_API_KEY>',
```

File: website/blog/2021-11-18-algolia-docsearch-migration/index.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+Please contact us via either [email](mailto:docsearch@algolia.com) or [DocSearch's Discord](https://discord.gg/bRTacwYrfX) or on [Docusaurus's Discord #algolia channel](https://discordapp.com/invite/docusaurus)
+
+### I have feedbacks!
+
+For any feedback regarding our documentation or the DocSearch UI component, you can open an issue [on our GitHub repository](https://github.com/algolia/docsearch/issues), but also contact us via the methods above.
+
+### Can I still run my own DocSearch instance?
+
+In favor of the new infrastructure and DocSearch v3, we will no longer maintain our beloved [DocSearch scraper](https://github.com/algolia/docsearch-scraper) and [DocSearch v2](https://github.com/algolia/docsearch/tree/master), but the repositories will still be available and open to pull requests.
```
Comment (bodinsamuel) [inline website/blog/2021-11-18-algolia-docsearch-migration/index.md:115]: Just personal feeling not strong suggestion
```suggestion
In favor of the new infrastructure and DocSearch v3, [DocSearch scraper](https://github.com/algolia/docsearch-scraper) and [DocSearch v2](https://github.com/algolia/docsearch/tree/master) will enter maintenance mode, those repositories will always be available and open to pull requests.
```

**Review by bodinsamuel — COMMENTED** [review]
Amazing 🙌🏻

File: website/blog/2021-11-18-algolia-docsearch-migration/index.md (hunk 6 above)
Comment (slorber) [reply website/blog/2021-11-18-algolia-docsearch-migration/index.md]: Probably quite implicit imho 😅

File: website/blog/2021-11-18-algolia-docsearch-migration/index.md (hunk 8 above)
Comment (slorber) [reply website/blog/2021-11-18-algolia-docsearch-migration/index.md:113]: That seems good enough to me and make it clearer that the scraper is not part of the "in favor" list of items.

This sentence is not very easy to read IMHO and could be totally rewritten (Hemingway app agrees).

But this blog post does not need to be 100% perfect either, so I'm going to move on and merge as is. Open to changes after publication if you feel it's important.

**Pull request merged**